lib/src/main/java/graphql/nadel/validation/NadelHydrationArgumentValidation.kt

```diff
@@ -0,0 +1,288 @@
+package graphql.nadel.validation
+
+import graphql.Scalars
+import graphql.nadel.dsl.RemoteArgumentDefinition
+import graphql.nadel.dsl.RemoteArgumentSource.SourceType.ObjectField
+import graphql.nadel.dsl.RemoteArgumentSource.SourceType.FieldArgument
+import graphql.nadel.dsl.UnderlyingServiceHydration
+import graphql.nadel.engine.util.isList
+import graphql.nadel.engine.util.isNonNull
+import graphql.nadel.engine.util.unwrapNonNull
+import graphql.nadel.engine.util.unwrapOne
+import graphql.schema.*
+
+internal class NadelHydrationArgumentValidation private constructor() {
+    companion object {
```

Comment:
This shouldn't be a companion object. Refer to the pattern used by the other classes.